fetcher/fetcher_test.go

```diff
@@ -154,9 +154,6 @@ func TestFetchSameKeysSlow(t *testing.T) {
 		go func(txID ids.ID, i int) {
 			defer wg.Done()
 			if i%2 == 0 {
-				// do some work to minimic timeout
-				for i := 0; i < 10000000; i++ {
-				}
 				close(delay)
```

Comment:
Wouldn't you want to do this after `f.Get` or at least after `f.Fetch`?

Would recommend writing out in comments what this `Slow` test is trying to do.